### ISSUE TITLE
Drawable should not depend on Container

### ISSUE THREAD

**Issue by UselessToucan** [ppy/osu-framework#102]
In Drawable class you can find:
`public Container Parent { get; set; }`
`internal void ChangeParent(Container parent){...}`

It means that Drawable class knows about Container and depends on it. From my point of view, it is not ok.
It can't be fixed just by replacing "Container" => "Drawable" in Drawable.cs, so I decided to open this issue.

**Comment by Tom94** [comment]
I feel like the best way to address this is trying to remove any upward-references in the draw hierarchy such that the `Parent` member is no longer needed. This entails figuring out an elegant way to invalidate `AutoSize` parents lazily, only with downward references.

Another plus of this approach would be trivial parallelization of the update traversal, as all children can be updated in parallel without data races.

**Comment by Tom94** [comment]
Let's decide whether we want to get rid of `Parent` sooner than later. (Maybe this weekend?)
We'd also need to discuss the logistics of making it work.

**Comment by Tom94** [comment]
Drawable now only has an `IContainer` which in turn does not rely on `Drawable`.
